packages/core/src/validators.ts

```diff
@@ -1039,6 +1071,18 @@ export class NotEmptyValidator extends Validator {
   }
 }
 
+export class NotEmptyObjectValidator extends Validator {
+  constructor(public readonly optionalProperties?: string[]) {
+    super();
+  }
+
+  validatePath(value: any, path: Path, ctx: ValidationContext): PromiseLike<ValidationResult> {
+    return !isNullOrUndefined(value) && !isEmptyObject(value)
+      ? ctx.successPromise(value)
+      : ctx.failurePromise(defaultViolations.notEmptyObject(value, path, this.optionalProperties), value);
```

Comment:
Optional properties are only used in error message? 🤔

Reply:
Hey @ssaarela as we discussed, will close this one and create a new PR!!!!